cpp/src/io/json/write_json.cu

```diff
@@ -504,6 +504,25 @@ struct column_to_strings_fn {
   {
   }
 
+  column_to_strings_fn(column_to_strings_fn const& other)
+    : options_{other.options_},
+      stream_{other.stream_},
+      mr_{other.mr_},
+      narep(other.narep, other.stream_),
+      struct_value_separator(other.struct_value_separator, other.stream_),
+      struct_row_begin_wrap(other.struct_row_begin_wrap, other.stream_),
+      struct_row_end_wrap(other.struct_row_end_wrap, other.stream_),
+      list_value_separator(other.list_value_separator, other.stream_),
+      list_row_begin_wrap(other.list_row_begin_wrap, other.stream_),
+      list_row_end_wrap(other.list_row_end_wrap, other.stream_)
+  {
+  }
```

Comment:
`column_to_strings_fn` is not meant to be copied. 
I just added the patch to fix the copy constructor from being invoked.

Reply:
```
diff --git a/cpp/src/io/json/write_json.cu b/cpp/src/io/json/write_json.cu
index c98e93aec4..98b35fc943 100644
--- a/cpp/src/io/json/write_json.cu
+++ b/cpp/src/io/json/write_json.cu
@@ -633,17 +633,17 @@ struct column_to_strings_fn {
 
     auto child_string_with_null = [&]() {
       if (child_view.type().id() == type_id::STRUCT) {
-        return (*this).template operator()<cudf::struct_view>(
+        return this->template operator()<cudf::struct_view>(
           child_view,
           children_names.size() > child_index ? children_names[child_index].children
                                               : std::vector<column_name_info>{});
       } else if (child_view.type().id() == type_id::LIST) {
-        return (*this).template operator()<cudf::list_view>(child_view,
+        return this->template operator()<cudf::list_view>(child_view,
                                                             children_names.size() > child_index
                                                               ? children_names[child_index].children
                                                               : std::vector<column_name_info>{});
       } else {
-        return cudf::type_dispatcher(child_view.type(), *this, child_view);
+        return cudf::type_dispatcher<cudf::id_to_type_impl, column_to_strings_fn const&>(child_view.type(), *this, child_view);
       }
     };
     auto new_offsets = cudf::lists::detail::get_normalized_offsets(
@@ -706,17 +706,17 @@ struct column_to_strings_fn {
                      auto const& current_col = thrust::get<1>(i_current_col);
                      // Struct needs children's column names
                      if (current_col.type().id() == type_id::STRUCT) {
-                       return (*this).template operator()<cudf::struct_view>(
+                       return this->template operator()<cudf::struct_view>(
                          current_col,
                          children_names.size() > i ? children_names[i].children
                                                    : std::vector<column_name_info>{});
                      } else if (current_col.type().id() == type_id::LIST) {
-                       return (*this).template operator()<cudf::list_view>(
+                       return this->template operator()<cudf::list_view>(
                          current_col,
                          children_names.size() > i ? children_names[i].children
                                                    : std::vector<column_name_info>{});
                      } else {
-                       return cudf::type_dispatcher(current_col.type(), *this, current_col);
+                       return cudf::type_dispatcher<cudf::id_to_type_impl, column_to_strings_fn const&>(current_col.type(), *this, current_col);
                      }
                    });
 
```

Reply:
```suggestion
  column_to_strings_fn(column_to_strings_fn const& other)      = delete;
```

Reply:
similarly other constructors and assignment operators can be deleted too.

Reply:
Thank you, @karthikeyann. The patch has been applied.

Reply:
TIL you can get the type dispatcher to take the functor by reference.
Thanks @karthikeyann!

Reply:
Very nice job on this, @karthikeyann. 👏